### PR TITLE
tests: settings: functional: nvs: adding native_posix target

### PR DESCRIPTION
**Pull request by kapi-no**
Added native_posix target to the functional tests of NVS.

Signed-off-by: Kamil Piszczek <Kamil.Piszczek@nordicsemi.no>

### CODE REVIEW

File: tests/subsys/settings/functional/nvs/prj_native_posix.conf [hunk 1]
```diff
@@ -0,0 +1,12 @@
+CONFIG_ZTEST=y
```
Comment (vanwinkeljan) [inline tests/subsys/settings/functional/nvs/prj_native_posix.conf]: Please create a `boards` directory and move board specific config in to this directory and keep one prj.conf file

File: tests/subsys/settings/functional/nvs/testcase.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
 tests:
   system.settings.functional.nvs:
-    platform_whitelist: qemu_x86
+    platform_whitelist: qemu_x86 native_posix
```
Comment (vanwinkeljan) [inline tests/subsys/settings/functional/nvs/testcase.yaml]: 64-bit support could be tested by adding `native_posix_64`  to the white list

File: tests/subsys/settings/functional/nvs/prj_native_posix.conf (hunk 1 above)
Comment (kapi-no) [reply tests/subsys/settings/functional/nvs/prj_native_posix.conf]: done, thanks for the tip

File: tests/subsys/settings/functional/nvs/testcase.yaml (hunk 2 above)
Comment (kapi-no) [reply tests/subsys/settings/functional/nvs/testcase.yaml]: added :)

**Review by rakons — APPROVED** [review]

**Review by Laczen — APPROVED** [review]
@vanwinkeljan, off PR topic: is creating a boards directory also a solution to keep overlays separated ? Can this also be done for libraries or modules (i.e can you specify a boards directory with overlays in a library or module to modify the standard board dts) ?

**Comment by carlescufi** [comment]
@kapi-no can you rebase?
@vanwinkeljan can you take another look?

**Comment by kapi-no** [comment]
> @kapi-no can you rebase?

done

**Review by vanwinkeljan — APPROVED** [review]

File: tests/subsys/settings/functional/nvs/boards/native_posix.conf [hunk 3]
```diff
@@ -0,0 +1 @@
+CONFIG_SETTINGS_NVS_SECTOR_SIZE_MULT=8192
```
Comment (MaureenHelm) [inline tests/subsys/settings/functional/nvs/boards/native_posix.conf]: Is this a configuration that should always be used for native posix? If so, set it in `boards/posix/native_posix/Kconfig.defconfig` instead?

**Comment by Laczen** [comment]
@kapi-no, I overlooked the sector size multiplier, this seems very high.

**Comment by Laczen** [comment]
> @Laczen, the NVS settings backend assumes that `DT_FLASH_ERASE_BLOCK_SIZE` (1 for `native_posix`) and the page size fetched from the Flash API (8192 for `native_posix`) are the same. The sector size is validated against page size during the NVS initialization.
> 
> I used the multiplier to align these values in the test, because I didn't want to touch `native_posix` DTS.

OK, thanks for the clarification.

@vanwinkeljan, any reason why DT_FLASH_ERASE_BLOCK_SIZE and page size (from flash API) are different ?

**Comment by vanwinkeljan** [comment]
> @vanwinkeljan, any reason why DT_FLASH_ERASE_BLOCK_SIZE and page size (from flash API) are different ?

@Laczen The native posix flash erase function does not check if the erase is a multiple of the sector size as such it is set to 1.

**Pull request merged**